diff --git a/node_modules/@enso-ui/charts/src/bulma/ChartCard.vue b/node_modules/@enso-ui/charts/src/bulma/ChartCard.vue
index 6a8d997..1965899 100644
--- a/node_modules/@enso-ui/charts/src/bulma/ChartCard.vue
+++ b/node_modules/@enso-ui/charts/src/bulma/ChartCard.vue
@@ -152,9 +152,9 @@ export default {
             this.$emit('fetching');
             this.loading = true;
             this.ongoingRequest?.cancel();
-            this.ongoingRequest = axios.CancelToken.source();
+            this.ongoingRequest = this.$axios.CancelToken.source();
 
-            axios.get(this.source, {
+            this.$axios.get(this.source, {
                 params: this.params,
                 cancelToken: this.ongoingRequest.token,
             }).then(({ data }) => {
diff --git a/node_modules/@enso-ui/charts/src/bulma/InfoPanel.vue b/node_modules/@enso-ui/charts/src/bulma/InfoPanel.vue
index 6d1fa8a..d085630 100644
--- a/node_modules/@enso-ui/charts/src/bulma/InfoPanel.vue
+++ b/node_modules/@enso-ui/charts/src/bulma/InfoPanel.vue
@@ -51,7 +51,6 @@
 <script>
 import ProgressCircle from '@enso-ui/progress-circle';
 import { clickOutside } from '@enso-ui/directives';
-import { shortNumber } from '@enso-ui/mixins';
 
 export default {
     name: 'InfoPanel',
@@ -111,7 +110,7 @@ export default {
 
     methods: {
         shortNumber(value) {
-            return shortNumber(value);
+            return this.$shortNumber(value);
         },
     },
 };
diff --git a/node_modules/@enso-ui/charts/src/components/Chart.vue b/node_modules/@enso-ui/charts/src/components/Chart.vue
index c4cb265..5bf8a20 100644
--- a/node_modules/@enso-ui/charts/src/components/Chart.vue
+++ b/node_modules/@enso-ui/charts/src/components/Chart.vue
@@ -1,6 +1,5 @@
 <script>
 import Chart from 'chart.js';
-import { shortNumber } from '@enso-ui/mixins';
 import 'chartjs-plugin-datalabels';
 import defaultOptions from './options';
 
@@ -72,14 +71,14 @@ export default {
 
             if (this.type !== 'bubble') {
                 options.plugins.datalabels.formatter = this.shortNumbers
-                    ? v => shortNumber(v)
+                    ? v => this.$shortNumber(v)
                     : this.formatter;
             }
 
             options.plugins.datalabels.display = this.display;
 
             if (options.scales) {
-                const callback = v => (this.shortNumbers ? shortNumber(v) : this.formatter(v));
+                const callback = v => (this.shortNumbers ? this.$shortNumber(v) : this.formatter(v));
 
                 options.scales.yAxes
                     .forEach(yAxis => (yAxis.ticks = { callback, ...yAxis.ticks }));
