diff --git a/node_modules/@enso-ui/select/src/renderless/CoreSelect.vue b/node_modules/@enso-ui/select/src/renderless/CoreSelect.vue
index c6e2c9b..2f80b47 100644
--- a/node_modules/@enso-ui/select/src/renderless/CoreSelect.vue
+++ b/node_modules/@enso-ui/select/src/renderless/CoreSelect.vue
@@ -263,10 +263,10 @@ export default {
                 this.ongoingRequest.cancel();
             }
 
-            this.ongoingRequest = axios.CancelToken.source();
+            this.ongoingRequest = this.$axios.CancelToken.source();
             this.loading = true;
 
-            axios.get(this.source, {
+            this.$axios.get(this.source, {
                 params: this.requestParams(),
                 cancelToken: this.ongoingRequest.token,
             }).then(({ data }) => {
@@ -276,7 +276,7 @@ export default {
                 this.loading = false;
             }).catch((error) => {
                 this.loading = false;
-                if (!axios.isCancel(error)) {
+                if (!this.$axios.isCancel(error)) {
                     this.errorHandler(error);
                 }
             });
