diff --git a/node_modules/@enso-ui/tables/src/renderless/CoreTable.vue b/node_modules/@enso-ui/tables/src/renderless/CoreTable.vue
index 3425092..eb18b05 100644
--- a/node_modules/@enso-ui/tables/src/renderless/CoreTable.vue
+++ b/node_modules/@enso-ui/tables/src/renderless/CoreTable.vue
@@ -207,7 +207,7 @@ export default {
             return this.state.filterScenarios.find(({ active }) => active);
         },
         init() {
-            axios.get(this.path, { params: { params: this.initParams } })
+            this.$axios.get(this.path, { params: { params: this.initParams } })
                 .then(({ data }) => {
                     this.prepare(data);
                     this.$nextTick(() => {
@@ -288,16 +288,16 @@ export default {
                 this.ongoingRequest.cancel();
             }
 
-            this.ongoingRequest = axios.CancelToken.source();
+            this.ongoingRequest = this.$axios.CancelToken.source();
 
             return this.template.method === 'GET'
-                ? axios[this.template.method.toLowerCase()](
+                ? this.$axios[this.template.method.toLowerCase()](
                     this.template.readPath, {
                         ...this.readRequest(this.template.method),
                         cancelToken: this.ongoingRequest.token,
                     },
                 )
-                : axios[this.template.method.toLowerCase()](
+                : this.$axios[this.template.method.toLowerCase()](
                     this.template.readPath,
                     this.readRequest(this.template.method),
                     { cancelToken: this.ongoingRequest.token },
@@ -331,7 +331,7 @@ export default {
             }).catch(error => {
                 this.meta.loading = false;
 
-                if (!axios.isCancel(error)) {
+                if (!this.$axios.isCancel(error)) {
                     this.errorHandler(error);
                 }
             });
@@ -419,7 +419,7 @@ export default {
                 : this.template.align;
         },
         exportData({ path, postEvent }) {
-            axios[this.template.method.toLowerCase()](
+            this.$axios[this.template.method.toLowerCase()](
                 path, this.readRequest(this.template.method, true),
             ).then(({ data }) => {
                 if (postEvent) {
@@ -431,7 +431,7 @@ export default {
             });
         },
         ajax(method, path, postEvent) {
-            axios[method.toLowerCase()](path).then(({ data }) => {
+            this.$axios[method.toLowerCase()](path).then(({ data }) => {
                 this.fetch();
 
                 if (postEvent) {
@@ -447,7 +447,7 @@ export default {
         }) {
             this.meta.loading = true;
 
-            axios[method.toLowerCase()](path, this.readRequest(method, false, selection))
+            this.$axios[method.toLowerCase()](path, this.readRequest(method, false, selection))
                 .then(({ data }) => {
                     if (postEvent) {
                         this.$emit(postEvent, data);
